src/snowflake/snowpark/_internal/compiler/query_generator.py

```diff
@@ -121,16 +122,23 @@ def do_resolve_with_resolved_children(
             # overwrite the SnowflakeCreateTable resolving, because the child
             # attribute will be pulled directly from the cache
             resolved_child = resolved_children[logical_plan.children[0]]
-            # when the plan is for SnowflakeCreateTable, there must be a snowflake_create_table_plan_info
-            # associated with the current query generator.
-            # TODO: this check need to be relaxed when large query breakdown with temp table
-            #       is implemented, because the child attributes are not necessary to create
-            #       the temp table
-            assert self._snowflake_create_table_plan_info is not None
-            assert (
-                self._snowflake_create_table_plan_info.table_name
+
+            # when creating a table during query compilation stage, if the
+            # table being created is the same as the one that is cached, we
+            # pull the child attributes directly from the cache. Otherwise, we
+            # use the child attributes as None. This will be for the case when
+            # table creation source is temp table from large query breakdown.
+            child_attributes = None
+            if (
+                logical_plan.creation_source
+                != TableCreationSource.LARGE_QUERY_BREAKDOWN
+                and self._snowflake_create_table_plan_info
+                and self._snowflake_create_table_plan_info.table_name
                 == logical_plan.table_name
-            )
+            ):
+                child_attributes = (
+                    self._snowflake_create_table_plan_info.child_attributes
```

Comment:
@sfc-gh-aalam how about the other comment for handling the creator in the create_query_generator here to only do the resolve if the creator is not large_query_breakdown https://github.com/snowflakedb/snowpark-python/blob/main/src/snowflake/snowpark/_internal/compiler/utils.py#L59